src/main/scala/chisel3/util/circt/PlusArgsTest.scala

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.circt
+
+import chisel3._
+import chisel3.experimental.{annotate, ChiselAnnotation, ExtModule}
+import chisel3.internal.Builder
+
+import circt.Intrinsic
+
+/** Create a module with a parameterized type which calls the verilog function
+  * $test$plusargs to test for the existance of the string str in the
+  * simulator command line.
+  */
+private class PlusArgsTestIntrinsic[T <: Data](gen: T, str: String) extends ExtModule(Map("FORMAT" -> str)) {
```

Comment:
why is gen: T , where T is any data, needed here? Should it just be a Bool type? can you show an `@example` of something where it fills out a useful Data of some arbitrary type here?